source/common/upstream/thread_aware_lb_impl.cc

```diff
@@ -116,13 +116,14 @@ void ThreadAwareLoadBalancerBase::refresh() {
     per_priority_state->global_panic_ = per_priority_panic_[priority];
 
     // Normalize host and locality weights such that the sum of all normalized weights is 1.
-    NormalizedHostWeightVector normalized_host_weights;
+    std::shared_ptr<NormalizedHostWeightVector> normalized_host_weights = std::make_shared<NormalizedHostWeightVector>();
     double min_normalized_weight = 1.0;
     double max_normalized_weight = 0.0;
-    normalizeWeights(*host_set, per_priority_state->global_panic_, normalized_host_weights,
+    normalizeWeights(*host_set, per_priority_state->global_panic_, *normalized_host_weights,
                      min_normalized_weight, max_normalized_weight);
     per_priority_state->current_lb_ =
-        createLoadBalancer(normalized_host_weights, min_normalized_weight, max_normalized_weight);
+        createLoadBalancer(*normalized_host_weights, min_normalized_weight, max_normalized_weight);
+    per_priority_state->normalized_host_weights_ = normalized_host_weights;
```

Comment:
nit: I think that we can just move `normalized_host_weights` to avoid atomic instructions for increment/decrement on shared pointer.

Reply:
This line is removed now.